code/bmpman/bmpman.cpp

```diff
@@ -2576,6 +2609,61 @@ void bm_page_in_start() {
 	gr_bm_page_in_start();
 }
 
+/**
+ * Stuffs Pathtypes with an additional path leading to a subdir of the same name as the given file (minus file extension).
+ * Used for allowing frames of .eff animations to be found in their own subdirectories.
+ *
+ * Example: if called with "debris.eff" and that file is in data/maps, then Pathtypes[CF_TYPE_TEMP_SUBDIR_LOOKUP].path
+ * will be stuffed with data/maps/debris, allowing data/maps/debris/debris_0000.dds (etc) to be found
+ */
+bool set_temp_subdir_pathtype(const char *filename) {
```

Comment:
I'm not sure what exactly you're suggesting; as far as I can tell, there's not really any expanding to do and this should work as-is for any other files. Of course it will still be limited to anchoring the subdirectory to an existing filename (effects/foo.eff <-> effects/foo/) but that's not something I think I could get rid of.

If for some reason one needed to use it for non-bitmaps, it would be easy at that point to move the function to cfile and remove the references to BMPMAN from the output strings. However, to me that seems like a rather unlikely scenario.